### ISSUE TITLE
stdout is not as buffered as it could be in some cases

### ISSUE THREAD

**Issue by alexcrichton** [rust-lang/rust#10279]
As part of #10270 I removed the page buffering schemed of a piped stdout in favor of a line buffering scheme. We will always want a line-buffering stdout at most for prints to a terminal, but when stdout is piped elsewhere we would ideally like to buffer more than just one line.

The reasoning for changing this is that diagnosing a runaway rust process becomes _much_ more difficult with a buffered stdout. Most programmers are expecting some sort of line buffer, but very few consider flushing stdout after all of their lines have been printed as well. As a result, a program killed by a signal currently doesn't flush stdout, meaning that all of that output is lost. This output is likely very important for diagnosing why the process went awry in the first place.

I would like to turn this buffering back on, but not until we have a good solution for this fairly common case.

**Comment by steveklabnik** [comment]
With IO reform looming, I'm not sure what the status is of this today.

**Comment by steveklabnik** [comment]
With no comments since my original one, I'm just going to close this. If anyone thinks it's still valid, please open a new issue with details, thanks!
